vicon/udp.h

```diff
@@ -228,14 +259,15 @@ class UDPClient
             m_ObjectData.resize(id + 1);
         }
 
-        // Update object data with translation and rotation
-        m_ObjectData[id].update(frameNumber, translation, rotation);
-
-        // Execute callback function, if set. Note that we copy by value here,
-        // which is presumably less efficient, but is thread safe
-        if (m_ReadCallback) {
-            m_ReadCallback(id, m_ObjectData[id], m_ReadUserData);
-        }
+        /*
+         * Update object data with position and attitude.
+         * 
+         * Note that we reorder the rotation angles we get from the Vicon system
+         * so that they are in the order of yaw, pitch and roll (which seems to
+         * be standard).
+         */
+        m_ObjectData[id].update(frameNumber, position[0], position[1], position[2],
+                                attitude[2], attitude[0], attitude[1]);
```

Comment:
Did you come round to the slightly weird templated object class system rather than the callbacks then?

Reply:
Yuh huh. I'm all about templating now as you may have noticed ;-)

Reply:
My work here is done :smile: